rpc/src/service_builder.rs

```diff
@@ -201,14 +202,20 @@ impl<'a> ServiceBuilder<'a> {
         db_config: &DBConfig,
         indexer_config: &IndexerConfig,
     ) -> Self {
-        let indexer = IndexerService::new(db_config, indexer_config, shared.async_handle().clone());
-        let indexer_handle = indexer.handle();
-        let rpc_methods = IndexerRpcImpl::new(indexer_handle).to_delegate();
-        if self.config.indexer_enable() {
-            start_indexer(&shared, indexer, indexer_config.index_tx_pool);
-            self.add_methods(rpc_methods);
-        } else {
-            self.update_disabled_methods("Indexer", rpc_methods);
+        match IndexerService::new(db_config, indexer_config, shared.async_handle().clone()) {
+            Ok(indexer) => {
+                let indexer_handle = indexer.handle();
+                let rpc_methods = IndexerRpcImpl::new(indexer_handle).to_delegate();
+                if self.config.indexer_enable() {
+                    start_indexer(&shared, indexer, indexer_config.index_tx_pool);
+                    self.add_methods(rpc_methods);
+                } else {
+                    self.update_disabled_methods("Indexer", rpc_methods);
+                }
+            }
+            Err(e) => {
+                warn!("Failed to enable indexer: {}", e);
```

Comment:
I believe the "error!" function should be used here. The CKB process is expected to terminate if the initialization of the indexer fails.

Reply:
In the case where the ckb node itself has not reached the height of the indexer's init_tip, the CKB should not exit.